Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -91,8 +91,8 @@ class OpenSearchEngine: NSObject, NSCoding {
     private func isSearchURLForEngine(url: NSURL?) -> Bool {
         guard let urlHost = url?.host,
             let queryEndIndex = searchTemplate.rangeOfString("?")?.startIndex,
-            let templateURL = NSURL(string: searchTemplate.substringToIndex(queryEndIndex)),
-            let templateURLHost = templateURL.host else { return false }
+            let templateURL = NSURL(string: searchTemplate.substringToIndex(queryEndIndex)) else { return false }
+        let templateURLHost = templateURL.extractDomainName()
```

Comment:
I'm looking at the implementation for `extractDomainName`, but I'm not exactly sure what it's supposed to do. Here's what I've found:

`www.google.com` becomes `google`
`search.yahoo.com` becomes `search.yahoo`
`google.co.uk` becomes `google.co`

These results don't seem very consistent to me -- sometimes the subdomain is stripped out (`www.`), other times it's not (`search.`); sometimes the full TLD is stripped away (`.com`), other times it's not (`.co.uk`).

Given the variability of domain names, I think string parsing alone won't be enough reliably to detect the TLD without more information. Have you tried something like `url.publicSuffixFromHost(url.host, withAdditionalParts: 0)`? (Not sure why this method takes a host since it's an extension on `NSURL`, but that's another bug). That uses our `effective_tld_names.dat` list to strip out known TLDs.